### DISCUSSION TITLE
404 responses with nextcloud-aio reverse proxy setup

### DISCUSSION THREAD

**Discussion by simwai** [nextcloud/all-in-one#4599] [Questions] [answered]
Hello, I tried to set up Nextcloud AIO in a reverse proxy setup. After days of trying around I finally decided to ask here now. I got far enough to start the containers from the initial panel on the Nextcloud website. When I open the dashboard the 404 responses start to happen, and the dashboard doesn't load appropriately. Basically, all ocs endpoint requests fail (s. below).

I have no firewall on my home lab server (which runs an Ubuntu container on Proxmox), because the connection from it to my public external VPS is established by ZeroTier VPN. On my public VPS the ports 80 and 443 for HTTP and HTTPS are open. 

Has anybody an idea if that is a bug or just a wrong nginx server config of me? I assume it is not a problem with my Docker Compose config, that's why I don't add it. If that's a wrong assumption, I am willing to share it too.

Every help here would be much appreciated, because I couldn't find an issue here where exactly that problem happened. I just found a kinda similar one, but the fix there didn't work for me.

Nginx config on public VPS server:
```nginx
location ~*
{
	proxy_pass http://172.25.100.143:11000$request_uri;

	proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
	proxy_set_header X-Forwarded-Port $server_port;
	proxy_set_header X-Forwarded-Scheme $scheme;
	proxy_set_header X-Forwarded-Proto $scheme;
	proxy_set_header X-Real-IP $remote_addr;
	proxy_set_header Accept-Encoding "";
	proxy_set_header Host $host;

	client_body_buffer_size 512k;
	proxy_read_timeout 86400s;
	client_max_body_size 0;

	# Websocket
	proxy_http_version 1.1;
	proxy_set_header Upgrade $http_upgrade;
	proxy_set_header Connection "upgrade";
}
```

Failed responses (three cases of dozens more):
```
Request URL:
https://cloud.wobbit.at/ocs/v2.php/apps/user_status/api/v1/heartbeat?format=json
Request Method:
PUT
Status Code:
404 Not Found
```

```
Request URL:
https://cloud.wobbit.at/ocs/v2.php/apps/notifications/api/v2/notifications
Request Method:
GET
Status Code:
404 Not Found
```

```
Request URL:
https://cloud.wobbit.at/ocs/v2.php/search/providers?from=%2Fapps%2Fdashboard%2F
Request Method:
GET
Status Code:
404 Not Found
```

![image](https://github.com/nextcloud/all-in-one/assets/16225108/fb914d52-4dcb-4e4c-afa9-4ac8cf8a3459)

**Comment by Zoey2936** [comment]
you location block seems to not match the recommended nginx config/location block: https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md#nginx-freenginx-openresty this seems to break websockets

**Reply by simwai** [reply]
I cannot use the template there, because I have Plesk. I just can set the stuff inside of the location.

**Reply by Zoey2936** [reply]
hmm then no idea, I can just tell that your websocket is broken

**Reply by simwai** [reply]
Hm okay thanks. I tried it now with the Apache config, but still no solution for me. :(

![image](https://github.com/nextcloud/all-in-one/assets/16225108/76cda332-0057-4521-b898-47b62d6448a7)

HTTP:
```Apache
RewriteEngine On
RewriteCond %{HTTPS} off
RewriteRule (.*) https://%{HTTP_HOST}%{REQUEST_URI}
RewriteCond %{SERVER_NAME} =cloud.wobbit.at
RewriteRule ^ https://%{SERVER_NAME}%{REQUEST_URI} [END,NE,R=permanent]
```

HTTPS:
```
RewriteEngine On
ProxyPreserveHost On
AllowEncodedSlashes NoDecode

ProxyPass / http://172.25.100.143:11000/ nocanon
ProxyPassReverse / http://172.25.100.143:11000/

RewriteCond %{HTTP:Upgrade} websocket [NC]
RewriteCond %{HTTP:Connection} upgrade [NC]
RewriteCond %{THE_REQUEST} "^[a-zA-Z]+ /(.*) HTTP/\d+(\.\d+)?$"
RewriteRule .? "ws://172.25.100.143:11000/%1" [P,L]

# Enable h2, h2c and http1.1
Protocols h2 h2c http/1.1

# Solves slow upload speeds caused by http2
H2WindowSize 5242880

# Disable HTTP TRACE method.
TraceEnable off
<Files ".ht*">
	Require all denied
</Files>

# Support big file uploads
LimitRequestBody 0

# TLS
SSLEngine               on
SSLProtocol             -all +TLSv1.2 +TLSv1.3
SSLCipherSuite          ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:DHE-RSA-CHACHA20-POLY1305
SSLHonorCipherOrder   off
SSLSessionTickets       off
```

**Comment by JoernSchoenyan** [comment, marked as answer]
Hello everyone,

to get Nextcloud AIO running together with Plesk, create a website in Plesk and disable PHP support for this domain. The Plesk configuration intercepts the connection because of the v2.php in the path. Then, as OP did, disable "Proxy mode" in the Nginx settings of that domain and use Additional nginx directives like this:

```
location / {
	proxy_pass http://127.0.0.1:11000$request_uri;
	proxy_http_version 1.1;

	proxy_set_header Upgrade $http_upgrade;
	proxy_set_header Connection "upgrade";

	proxy_set_header Host $host;
	proxy_set_header X-Real-IP $remote_addr;
	proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
	proxy_set_header X-Forwarded-Proto $scheme;
	proxy_set_header X-Forwarded-Port $server_port;
	proxy_set_header X-Forwarded-Scheme $scheme;
	proxy_set_header Early-Data $ssl_early_data;
}
```
